### PR TITLE
[core] Add convenience method `RAY_LOG(INFO).WithField(job_id)`.

### PR DESCRIPTION
**Pull request by rynewang**
We added structured logging `RAY_LOG(INFO).WithField(kLogKeyJobID, job_id)`. However this is still a bit cumbersome and error prone, since one can write kLogKeyJobID with a NodeID and so on. This PR introduces `WithField(job_id)` and detects the field key by type, saving hassle and error.

Also added a new key for ObjectID.

As an example, changed all WithField in `gcs_actor_manager.cc`.

### CODE REVIEW

**Review by jjyao — APPROVED** [review]
Nice

**Comment by rynewang** [comment]
auto merge?

**Comment by jjyao** [comment]
microcheck failure

**Comment by rynewang** [comment]
can't do this - id.h depends on logging.h, which can't depend on id.h. Tried to forward declare but also failed.

**Comment by rynewang** [comment]
Trying an approach with traits

**Review by jjyao — APPROVED** [review]

File: src/ray/util/logging.h [hunk 1]
```diff
@@ -314,6 +320,11 @@ class RayLog {
     }
   }
 
+  template <typename T, typename enable = decltype(DefaultLogKey<T>::key)>
```
[the remark this answers is not in the document]
Comment (rynewang) [reply src/ray/util/logging.h:323]: it needs not. Updated

**Review by jjyao — APPROVED** [review]

**Pull request merged**